sbi/neural_nets/ratio_estimators.py

```diff
@@ -0,0 +1,93 @@
+from abc import ABC, abstractmethod
+
+import torch
+from torch import Tensor, nn
+
+
+class RatioEstimator(nn.Module, ABC):
+    r"""Base class for ratio estimators.
+
+    The ratio estimator class is a wrapper around neural networks that enables
+    evaluation of `unnormalized_log_ratio` for `theta`, `x` pairs. It also
+    provides a method for combining `theta` and `x` into a single tensor.
+
+    Note:
+        We assume that the input to the ratio estimator is a tensor of shape
+        (sample, batch, event) where sample are iid draws, batch are
+        independent (not necessarily identically distributed) draws, and
+        event can be totally dependent.
+        See: https://bochang.me/blog/posts/pytorch-distributions/
+    """
+
+    def __init__(
+        self,
+        net: nn.Module,
```

Comment:
`DensityEstimator`s all get `condition_shape` as input and I will also soon add `input_shape` (the event shapes), see [here](https://github.com/sbi-dev/sbi/pull/1138). Please add them also to the init and save them as attributes `self.condition_shape` and `self.input_shape`

Reply:
Feel free to call them `x_shape` and `theta_shape` here.

Reply:
okay I entered them, but supporting `sample_dim`  will take more time.

Reply:
now it just checks for (1, 1, event) or (1, event) or (event)